CMakeLists.txt

```diff
@@ -209,6 +209,41 @@ else()
   endif()
 endif()
 
+find_package(hip)
+
+if(hip_FOUND)
+  get_filename_component(CXX_COMPILER ${CMAKE_CXX_COMPILER} NAME)
+  if(hip_VERSION VERSION_LESS "5.3")
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      message(WARNING " Cannot use gcc for rocm < 5.3")
+      unset(HAVE_ROCSPARSE)
```

Comment:
This does not work, `HAVE_ROCSPARSE` is still 1. Setting it to 0 works, but it is still defined as `HAVE_ROCSPARSE 0` in `config.h`.

When testing with rocblas enabled, rocsparse is found, and a linker error occurs. This is because `rocsparseSolverBackend.cpp` does not get compiled, because `rocblas_FOUND` is false. But since `HAVE_ROCSPARSE` is 1, the `BdaBridge` includes `rocsparseSolverBackend.hpp`.